README.md

```diff
@@ -61,13 +61,19 @@ Package [samza.examples.cookbook](https://github.com/apache/samza-hello-samza/tr
 Package [samza.examples.wikipedia.application](https://github.com/apache/samza-hello-samza/tree/master/src/main/java/samza/examples/wikipedia/application) contains a small Samza application which consumes the real-time feeds from Wikipedia, extracts the metadata of the events, and calculates statistics of all edits in a 10-second window. You can start the app on the grid using the run-app.sh script:
 
 ```
-./deploy/samza/bin/run-app.sh --config-factory=org.apache.samza.config.factories.PropertiesConfigFactory --config-path=file://$PWD/deploy/samza/config/wikipedia-application.properties
+./deploy/samza/bin/run-app.sh \
+  --config app.class=samza.examples.wikipedia.application.WikipediaApplication \
+  --config yarn.package.path=file:///Users/kwu/workspace/hello-samza/target/hello-samza-1.5.0-SNAPSHOT-dist.tar.gz \
+  --config job.name=wikipedia-application \
+  --config job.factory.class=org.apache.samza.job.yarn.YarnJobFactory \
```

Comment:
These all seem to be copied from the properties file. It seems like it would be non-trivial to keep the properties file and this list of configs consistent, since they are in different places.
https://github.com/apache/samza/pull/1256 indicates that you only need the config loader and properties path, but here you are adding several other configs.

Reply:
submission related configs, such as job.name, app.class, job.factory.class and yarn.package.path needs to be explicitly passed in during submission since we are not reading config files anymore during submission.

I am cleaning up properties file ATM as well.

I believe the previous PR missed the instruction that all submission related configs needs to be provided explicitly.

Reply:
properties files cannot be simplified as they may also be used for standalone deployment where we do not need to feed the submission config.

Reply:
Do you have a clean way to describe how to specify submission configs in general? Based on your comments, it seems like standalone (`LocalApplicationRunner`) specifies a different set of submission configs than YARN (`RemoteApplicationRunner`), even though some of those configs are general to Samza (e.g. `app.class`).
It would be good to have as few runner-specific steps as possible. `ApplicationRunner` is the interface, so it would be nice to not have to worry about the specific `ApplicationRunner` being used when trying to start the app. I admit that Samza does already do some environment-specific configs (e.g. YARN-specific configs are needed when using `YarnJobFactory`), but we should generally minimize that.
I'm not sure if this works, but could we recommend standalone to pass the larger set of submission configs (similar to YARN) also? Then there would be more consistency. It would be easier to describe what submission configs are and how to specify them in general.